op-supervisor/supervisor/backend/syncnode/reset.go

```diff
@@ -200,11 +210,32 @@ func (t *resetTracker) resetHeadsFromTarget(target eth.BlockID) {
 		return
 	}
 
+	// TODO(#16026): This function should be split off from the resetTracker
 	t.managed.log.Info("reset target identified", "target", target)
 	var lUnsafe, xUnsafe, lSafe, xSafe, finalized eth.BlockID
 
-	// the unsafe block is always the last block we found to be consistent
-	lUnsafe = target
+	// Check if target is seen in unsafe DB.
+	// If it is, the tip of the unsafe DB is good to use as unsafe-reset-target.
+	// If it is not, then the unsafe logs DB is inconsistent with our local-safe DB.
+	// So we need to stick to the target block.
+	if err := t.managed.backend.IsLocalUnsafe(internalCtx, t.managed.chainID, target); err != nil {
+		if errors.Is(err, types.ErrConflict) {
+			lUnsafe = target
+		} else {
+			t.managed.log.Error("Failed to check local-unsafe", "target", target, "err", err)
+			t.endReset()
+			return
+		}
+	} else {
+		// TODO(#16026): cleanup how we fall back to the tip of unsafe chain
```

Comment:
what's meant by this comment?

Reply:
I'll clarify in the github issue that we want rewind-locking around this function that determines the reset continuation point.